### PR TITLE
Fix generation when class params has `val` keyword

### PR DESCRIPTION
**Pull request by andrelfpinto**

### CODE REVIEW

File: rules/src/main/scala/fix/GenerateDataClass.scala [hunk 1]
```diff
@@ -183,7 +183,7 @@ class GenerateDataClass(config: Configuration) extends SemanticRule("GenerateDat
     def cleanParam(param: Term.Param): Term.Param =
       param.copy(
         default = None,
-        mods = param.mods.filterNot(m => m.toString.startsWith(sinceStr)),
+        mods = Nil,
```
Comment (hamnis) [inline rules/src/main/scala/fix/GenerateDataClass.scala:186]: why was this removed?

Comment (andrelfpinto) [reply rules/src/main/scala/fix/GenerateDataClass.scala:186]: Let's say someone writes this:

```scala
@data class Foo(@since private final val bar: Int)
```

When `scalafix` generates the code, we should get this:

```scala
object Foo {
  def apply(bar: Int): Foo = new Foo(bar)
}
```

Not only `Mod.Annot` should be removed but the other modifiers too. The original motivation was to remove `Mod.ValParam`, but the others can't be in the `apply` method either.

**Review by hamnis — APPROVED** [review]
